src/ServerClient/ServerClient.ts

```diff
@@ -401,6 +423,81 @@ class ServerClient {
         let retryDelay = (Math.exp(retryNumber - 1) + Math.random()*2) * 1000;
         return setTimeout(method, retryDelay);
     }
+
+    private createBatchPostPromiseFromXhr = (url, payload, token, responseTextFormat, onProgressChange = (percentComplete) => {}, batchSize = 2, maxByteSize = 8000000) => {
```

Comment:
This function is wild.  Big nested-recursive-promise vibes.  Equal parts terrifying & impressive.

Reply:
Ahh, I just noticed the batch size is still to small (had set it for testing purpose)! :) Reverting it back to 10000 as it was before, thanks!

Reply:
yeah i actually think we could break out some of the logic inside postBatch into smaller functions defined above, just so the logic of postbatch is more clear.  kind of hard to understand where it starts and begins, and what resolve(postBatch()) even means right now